### PR TITLE
DC-251: Add source clear and sonarqube scanners to build

### PR DESCRIPTION
**Pull request by pshapiro4broad**
This adds source clear and sonarqube scans to our PR tests. Source clear checks for vulnerabilities in .jars that we depend on, and sonarqube runs a number of static code analysis checks on our Java code.

~Note that builds will fail until the GitHub secrets are published once https://github.com/broadinstitute/terraform-ap-deployments/pull/551 is merged.~

### CODE REVIEW

File: .github/workflows/build-and-test.yml [hunk 1]
```diff
@@ -112,6 +112,54 @@ jobs:
...
+      - name: Gradle cache
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.gradle/caches
+            ~/.gradle/wrapper
+          key: v1-${{ runner.os }}-gradle-${{ github.ref }}-${{ github.sha }}
+
+      - name: SonarQube scan
+        run: ./gradlew -Dsonar.login="${{ secrets.SONAR_TOKEN }}" --build-cache sonarqube
```
Comment (samanehsan) [inline .github/workflows/build-and-test.yml:161]: I'm getting confused about the global vs. service build.gradle files. If the sonarqube task is defined in the service, why isn't this :service:sonarqube? 🤔

File: buildSrc/build.gradle [hunk 2]
```diff
@@ -12,10 +12,12 @@ repositories {
 dependencies {
     implementation 'com.diffplug.spotless:spotless-plugin-gradle:6.3.0'
     implementation 'com.google.cloud.tools.jib:com.google.cloud.tools.jib.gradle.plugin:3.2.0'
+//    implementation 'com.srcclr.gradle:com.srcclr.gradle.gradle.plugin:3.1.10'
```
[the remark this answers is not in the document]
Comment (pshapiro4broad) [reply buildSrc/build.gradle]: Yes, forgot to delete that.

File: .github/workflows/build-and-test.yml (hunk 1 above)
Comment (pshapiro4broad) [reply .github/workflows/build-and-test.yml:161]: Either would work in this case. The task is only run in the subprojects that have it present, so using a global task means that if we add this scan to other projects, we don't have to update the Gradle command. This seems like a nice feature to me, but I could make it more specific instead if you think it's confusing.

Comment (samanehsan) [reply .github/workflows/build-and-test.yml:161]: Thanks for clarifying! I think we can leave it as-is so we don't have to update the command later

File: .github/workflows/build-and-test.yml [hunk 3]
```diff
@@ -112,6 +112,54 @@ jobs:
...
+          java-version: '17'
+          distribution: 'temurin'
+
+      - name: Gradle cache
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.gradle/caches
+            ~/.gradle/wrapper
+          key: v1-${{ runner.os }}-gradle-${{ github.ref }}-${{ github.sha }}
```
Comment (fboulnois) [inline .github/workflows/build-and-test.yml:133]:
```suggestion
          distribution: 'temurin'
          cache: 'gradle'
```

File: .github/workflows/build-and-test.yml [hunk 4]
```diff
@@ -112,6 +112,54 @@ jobs:
...
```
Comment (fboulnois) [inline .github/workflows/build-and-test.yml:158]:
```suggestion
          distribution: 'temurin'
          cache: 'gradle'
```

File: .github/workflows/build-and-test.yml (hunk 1 above)
Comment (fboulnois) [inline .github/workflows/build-and-test.yml:161]:
```suggestion
        run: ./gradlew --build-cache -Dsonar.login="${{ secrets.SONAR_TOKEN }}" sonarqube
```

nit: lead with `--build-cache` for consistency with other items?

**Review by fboulnois — CHANGES_REQUESTED** [review]
A few suggestions:

File: .github/workflows/build-and-test.yml (hunk 1 above)
Comment (fboulnois) [reply .github/workflows/build-and-test.yml:161]: Also, does the code need to be containerized beforehand for any of these tools to work correctly?

**Review by fboulnois — APPROVED** [review]
lgtm

**Review by samanehsan — APPROVED** [review]

**Pull request merged**